pkg/virt-controller/watch/migration_test.go

```diff
@@ -1039,6 +1039,28 @@ var _ = Describe("Migration watcher", func() {
 			// expect nothing to occur
 		})
 
+		It("should not transition to PreparingTarget if VMI MigrationState is outdated", func() {
```

Comment:
I am a little bit lost. Can you explain how the test works and if the name is correct?

Reply:
The name is correct, the test check the correctness of the fix introduced in commit 98bc19d13b3ccbec6efea9e73d8f6bbfb8d78a89.

Basically, it checks that the controller moves the migration state from `Scheduled` to `PreparingTarget` only if the `MigrationState` field inside the VMI object has been updated after the target handoff.

Reply:
Reading the test I have a hard time understanding which part should reflect `only if the MigrationState field inside the VMI object has been updated after the target handoff.` Can you point me to it?

Reply:
https://github.com/kubevirt/kubevirt/blob/814979ccd696c6a38bd972e4408649c114fa7946/pkg/virt-controller/watch/migration_test.go#L1051

I'm populating the VMI migrationState with "oldmigrationuid" which is different than the actual new migration uid https://github.com/kubevirt/kubevirt/blob/814979ccd696c6a38bd972e4408649c114fa7946/pkg/virt-controller/watch/migration_test.go#L1045 "testmigration".